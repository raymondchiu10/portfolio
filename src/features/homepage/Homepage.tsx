import About from "./components/About";
import { Hero } from "./components/Hero";

export function Homepage() {
	return (
		<section className="homepage">
			<Hero />
			<About />
		</section>
	);
}
