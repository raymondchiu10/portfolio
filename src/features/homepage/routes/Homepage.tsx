import { Hero } from "../components/Hero";
import About from "../components/About";

export function Homepage() {
	return (
		<section className="homepage">
			<Hero />
			<About />
		</section>
	);
}
