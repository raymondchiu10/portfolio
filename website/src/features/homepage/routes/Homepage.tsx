import { Header } from "../components/Header";

export function Homepage() {
	return (
		<>
			<Header />
		</>
	);
}