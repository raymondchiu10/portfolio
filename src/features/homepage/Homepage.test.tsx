import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Homepage from "./Homepage";

describe("home page render", () => {
	it("renders the page", () => {
		const { asFragment } = render(<Homepage />);
		expect(asFragment()).toMatchSnapshot();
	});
});
