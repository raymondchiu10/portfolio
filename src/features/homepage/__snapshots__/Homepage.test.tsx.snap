// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`home page render > renders the page 1`] = `
<DocumentFragment>
  <section
    class="homepage"
  >
    <section
      class="hero"
    >
      <div
        class="hero_container"
      >
        <h2
          class="hero_container_header"
        >
          Hey there, I'm Raymond
        </h2>
        <p
          class="hero_container_subheader"
        >
          I'm a Fullstack Developer who enjoys frontend development and bringing experiences and ideas to life.
        </p>
      </div>
    </section>
    <section
      class="about"
    >
      <div
        class="about_container"
      >
        <h3
          class="about_container_header"
        >
          Projects I have worked on:
        </h3>
      </div>
      <div
        class="about_carousel"
      >
        <div
          class="carousel-root"
        >
          <div
            class="carousel carousel-slider"
            style="width: 100%;"
          >
            <button
              aria-label="previous slide / item"
              class="control-arrow control-prev"
              type="button"
            />
            <div
              class="slider-wrapper axis-horizontal"
            >
              <ul
                class="slider animated"
                style="transform: translate3d(-70%,0,0); transition-duration: 350ms;"
              >
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/xai_games_sentrynodes_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/xai_games_sentrynodes_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/xai games sentrynodes screenshot"
                        src="/src/images/jpg/xai_games_sentrynodes_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide selected previous"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/frameonesoftware_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/frameonesoftware_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/frameonesoftware screenshot"
                        src="/src/images/jpg/frameonesoftware_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/frameonesoftware_screenshot2.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/frameonesoftware_screenshot2.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/frameonesoftware screenshot2"
                        src="/src/images/jpg/frameonesoftware_screenshot2.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/expopulus_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/expopulus_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/expopulus screenshot"
                        src="/src/images/jpg/expopulus_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/expopulus_dags_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/expopulus_dags_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/expopulus dags screenshot"
                        src="/src/images/jpg/expopulus_dags_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/expopulus_finalform_carousel_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/expopulus_finalform_carousel_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/expopulus finalform carousel screenshot"
                        src="/src/images/jpg/expopulus_finalform_carousel_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/expopulus_ironpigeons_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/expopulus_ironpigeons_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/expopulus ironpigeons screenshot"
                        src="/src/images/jpg/expopulus_ironpigeons_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/expopulus_ironpigeons_screenshot2.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/expopulus_ironpigeons_screenshot2.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/expopulus ironpigeons screenshot2"
                        src="/src/images/jpg/expopulus_ironpigeons_screenshot2.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/rxtome_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/rxtome_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/rxtome screenshot"
                        src="/src/images/jpg/rxtome_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/xai_games_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/xai_games_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/xai games screenshot"
                        src="/src/images/jpg/xai_games_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/xai_games_sentrynodes_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/xai_games_sentrynodes_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/xai games sentrynodes screenshot"
                        src="/src/images/jpg/xai_games_sentrynodes_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
                <li
                  class="slide selected previous"
                  style="min-width: 80%;"
                >
                  <div
                    class="about_carousel_item"
                  >
                    <picture>
                      <source
                        srcset="/src/images/webp/frameonesoftware_screenshot.webp"
                        type="image/webp"
                      />
                      <source
                        srcset="/src/images/jpg/frameonesoftware_screenshot.jpg"
                        type="image/jpeg"
                      />
                      <img
                        alt="jpg/frameonesoftware screenshot"
                        src="/src/images/jpg/frameonesoftware_screenshot.jpg"
                      />
                    </picture>
                  </div>
                </li>
              </ul>
            </div>
            <button
              aria-label="next slide / item"
              class="control-arrow control-next"
              type="button"
            />
            <p
              class="carousel-status"
            >
              1 of 10
            </p>
          </div>
          <div
            class="carousel"
          >
            <div
              class="thumbs-wrapper axis-vertical"
            >
              <button
                aria-label="previous slide / item"
                class="control-arrow control-prev control-disabled"
                type="button"
              />
              <ul
                class="thumbs animated"
                style="transform: translate3d(0,0,0); transition-duration: 350ms;"
              />
              <button
                aria-label="next slide / item"
                class="control-arrow control-next control-disabled"
                type="button"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
